executor/binary.go

```diff
@@ -1,27 +1,91 @@
 package executor
 
 import (
-	"os/exec"
+	"fmt"
+	"os"
+	"path/filepath"
 
+	"github.com/screwdriver-cd/sd-cmd/config"
 	"github.com/screwdriver-cd/sd-cmd/screwdriver/api"
+	"github.com/screwdriver-cd/sd-cmd/screwdriver/store"
 )
 
 // Binary is a Binary Executor object
 type Binary struct {
-	Cmd *api.Command
-	Arg []string
+	Args    []string
+	Store   store.Store
+	Command *store.Command
 }
 
 // NewBinary returns Binary object
-func NewBinary(cmd *api.Command, arg []string) (*Binary, error) {
+func NewBinary(spec *api.Command, arg []string) (*Binary, error) {
+	storeapi, err := store.New(spec)
+	if err != nil {
+		return nil, err
+	}
 	binary := &Binary{
-		Cmd: cmd,
-		Arg: arg,
+		Args:  arg,
+		Store: storeapi,
 	}
 	return binary, nil
 }
 
+func (b *Binary) download() error {
```

Comment:
We should avoid using single character variables. Because we can not search it easily.

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#variable-names

Golang prefer to use short character variables. I think to use single character for receiver is good for golang project.

Reply:
I got the convention about a method receiver. Thanks.